File: VB/Model/DataHelper.vb
Imports System.Collections.ObjectModel

Namespace WpfApplication

    Public Class DataHelper

        Public Shared Function GetDataSource(ByVal count As Integer) As Object
            Dim collection As ObservableCollection(Of GridItem) = New ObservableCollection(Of GridItem)()
            For i As Integer = 0 To count - 1
                collection.Add(New GridItem(Date.Now.AddMinutes(count * i).AddDays((i - count \ 2) * count), String.Format("Name{0}", i), i))
            Next

            Return collection
        End Function
    End Class

    Public Class GridItem

        Public Sub New(ByVal [date] As Date, ByVal name As String, ByVal iD As Integer)
            Me.Date = [date]
            Me.Name = name
            Me.ID = iD
        End Sub

        Public Property [Date] As Date

        Public Property Name As String

        Public Property ID As Integer
    End Class
End Namespace

File: CS/Model/DataHelper.cs
using System;
using System.Collections.ObjectModel;

namespace WpfApplication {
    public class DataHelper
    {
        public static object GetDataSource(int count)
        {
         ObservableCollection<GridItem> collection = new ObservableCollection<GridItem>();
            for (int i = 0; i < count; i++)
            {
                collection.Add(new GridItem(DateTime.Now.AddMinutes(count * i).AddDays((i - count / 2) * count), String.Format("Name{0}", i), i));
            }
            return collection;
        }
  
    }

    public class GridItem {
        public GridItem(DateTime date, string name, int iD) {
            Date = date;
            Name = name;
            ID = iD;
        }

        public DateTime Date { get; set; }
        public string Name { get; set; }
        public int ID { get; set; }
    }
}